### PR TITLE
[Fix bug 1248893] Don't force new curators in legacy groups edit.

### PR DESCRIPTION
**Pull request by johngian**
* Fix initial data in curator form.
* When legacy group, initial should be empty.

### CODE REVIEW

**Comment by johngian** [comment]
@akatsoulas r?

File: mozillians/groups/forms.py [hunk 1]
```diff
@@ -47,7 +49,11 @@ def clean(self):
                 self._errors['new_member_criteria'] = self.error_class([msg])
                 del cleaned_data['new_member_criteria']
 
-        if not curators:
+        # Check if group is a legacy group without curators.
+        # In this case we should allow curators relation to be empty.
+        allow_no_curators = self.instance.pk and not self.instance.curators.exists()
```
Comment (akatsoulas) [inline mozillians/groups/forms.py]: I would go with `group_has_curators` and remove the `not` both here and a couple of lines below. It's really minor but I think it helps the readability of the code. nice catch btw!

File: mozillians/groups/views.py [hunk 2]
```diff
@@ -397,7 +397,8 @@ def group_add_edit(request, url=None):
 
     form_class = SuperuserGroupForm if is_manager else GroupForm
 
-    curators_ids = [profile.id]
+    curators_ids = [] if group.pk and not group.curators.exists() else [profile.id]
```
Comment (akatsoulas) [inline mozillians/groups/views.py]: Again here lets reverse the order: [profile.id] if curators.exists() else []

File: mozillians/groups/tests/test_forms.py [hunk 3]
```diff
@@ -41,3 +42,33 @@ def test_no_saved_criteria(self):
         form = GroupForm(data=form_data)
         ok_(form.is_valid())
         eq_(u'', form.cleaned_data['new_member_criteria'])
+
+    def test_legacy_group_curators_validation(self):
```
Comment (akatsoulas) [inline mozillians/groups/tests/test_forms.py]: nice :sun_with_face:

**Comment by akatsoulas** [comment]
code looks good. Haven't tested this though. r+wc

**Pull request merged**